### PR TITLE
test tanner-parse-response

### PR DESCRIPTION
**Pull request by rjt-gupta**
Added test for `parse_tanner_response`.
Increased test coverage of `tanner_handler.py`  to 92%.
Refs - #187

### CODE REVIEW

File: snare/tests/test_parse_tanner_response.py [hunk 1]
```diff
@@ -73,6 +73,29 @@ def test_parse_type_two(self):
...
+        }
+
+        async def test():
+            (self.res1, self.res2,
+             self.res3, self.res4) = await self.handler.parse_tanner_response(self.requested_name, self.detection)
+
+        self.loop.run_until_complete(test())
+        expected_header = {"content-type": "multipart/form-data"}
+        real_result = [self.res1, self.res2, self.res3, self.res4]
+        expected_result = [b'test.png', 'image/png', expected_header, 200]
```
Comment (rnehra01) [inline snare/tests/test_parse_tanner_response.py]: Use variables `self.expected_content`, `self.content_type` and assign after `detection`. Also I think to maintain consistency we can make `self.expected_header` and `self.expected_status_code` in setUp method.

**Comment by rnehra01** [comment]
@rjt-gupta Can you try to add the remaining cases to make coverage 100%?

**Comment by rjt-gupta** [comment]
Yes, Im working on it.

**Comment by rjt-gupta** [comment]
The expected status code is `200` throughout the test. I don't think we should `self.expected_sataus_code` in setUp method. what do you think? @rnehra01

**Comment by rnehra01** [comment]
Look [here](https://github.com/mushorg/snare/blob/master/snare/tanner_handler.py#L88).

**Comment by rjt-gupta** [comment]
Oh yes, I missed this part as this part is not covered in the test yet. Will do the change surely :)

**Comment by rjt-gupta** [comment]
Done.

File: snare/tests/test_parse_tanner_response.py [hunk 2]
```diff
@@ -54,6 +56,36 @@ def test_parse_type_one(self):
...
+             self.res3, self.res4) = await self.handler.parse_tanner_response(self.requested_name, self.detection)
+
+        self.loop.run_until_complete(test())
+        real_result = [self.res1, self.res2, self.res3, self.res4]
+        expected_result = [self.page_content, self.content_type, {}, 200]
+        self.assertCountEqual(real_result, expected_result)
+
+    def test_parse_type_one_error(self):
+        self.detection = {"type": 1}
+        meta_content = {"/index.html": {}}
```
Comment (rnehra01) [inline snare/tests/test_parse_tanner_response.py]: This can be removed.

File: snare/tests/test_parse_tanner_response.py [hunk 3]
```diff
@@ -54,6 +56,36 @@ def test_parse_type_one(self):
...
+        self.requested_name = 'something/'
+        self.handler = TannerHandler(self.args, meta_content, self.uuid)
```
Comment (rnehra01) [inline snare/tests/test_parse_tanner_response.py]: This can be removed too.

File: snare/tests/test_parse_tanner_response.py [hunk 4]
```diff
@@ -105,17 +185,5 @@ def test_call_handle_html(self):
         self.loop.run_until_complete(test())
         self.handler.html_handler.handle_content.assert_called_with(self.call_content)
 
-    def test_parse_exception(self):
```
Comment (rnehra01) [inline snare/tests/test_parse_tanner_response.py]: Why did you remove this?

File: snare/tests/test_parse_tanner_response.py [hunk 5]
```diff
@@ -73,6 +105,54 @@ def test_parse_type_two(self):
...
+        real_result = [self.res1, self.res2, self.res3, self.res4]
+        expected_result = [self.expected_content, self.content_type, self.expected_header, 200]
+        self.assertCountEqual(real_result, expected_result)
+
+    def test_parse_type_two_error(self):
+
+        self.detection = {
+            "type": 2,
+            "payload": {
+                "page": "/index.html",
```
Comment (rnehra01) [inline snare/tests/test_parse_tanner_response.py]: In this case also you can simulate the error using 
```suggestion
                "page": "/something",
```
and remove line [142](https://github.com/mushorg/snare/pull/192/files#diff-5a32a1b31bb3574ea82f22145d7864a7R142) and line 143

File: snare/tests/test_parse_tanner_response.py [hunk 6]
```diff
@@ -73,6 +105,54 @@ def test_parse_type_two(self):
...
+                "value": "test"
+            }
+        }
+        meta_content = {"/index.html": {}}
+        self.handler = TannerHandler(self.args, meta_content, self.uuid)
+        self.expected_content = b'<html><body><div>test</div></body></html>'
+        self.content_type = r'text\html'
```
Comment (rnehra01) [inline snare/tests/test_parse_tanner_response.py]: I think it should be 
```suggestion
        self.content_type = r'text/html'
```
Right?
And can you first make a separate PR to fix this https://github.com/mushorg/snare/blob/c04984110c15d8fbe031a77d51af7f17a423ca5b/snare/tanner_handler.py#L109

File: snare/tests/test_parse_tanner_response.py (hunk 4 above)
Comment (rjt-gupta) [reply snare/tests/test_parse_tanner_response.py]: The coverage is reached using other functions so need for this one.

File: snare/tests/test_parse_tanner_response.py (hunk 6 above)
Comment (rjt-gupta) [reply snare/tests/test_parse_tanner_response.py]: Roger!

File: snare/tests/test_parse_tanner_response.py (hunk 2 above)
Comment (rjt-gupta) [reply snare/tests/test_parse_tanner_response.py]: Yes, even though I have simulated the test-case using `something/` but there is no test that demonstrates this case which is also a possibility. 
But your decision would be final, should I remove this?

File: snare/tests/test_parse_tanner_response.py (hunk 6 above)
Comment (rjt-gupta) [reply snare/tests/test_parse_tanner_response.py]: PR submitted.

**Comment by rjt-gupta** [comment]
Coverage is at 99%. How should I test this [line](https://github.com/mushorg/snare/blob/c04984110c15d8fbe031a77d51af7f17a423ca5b/snare/tanner_handler.py#L58).

File: snare/tests/test_parse_tanner_response.py (hunk 2 above)
Comment (rnehra01) [reply snare/tests/test_parse_tanner_response.py]: This case need not be handled here, it can be checked in `cloner.py`. So I think it's good to keep it simple here.

File: snare/tests/test_parse_tanner_response.py (hunk 4 above)
Comment (rnehra01) [reply snare/tests/test_parse_tanner_response.py]: I think this test is for [this](https://github.com/mushorg/snare/blob/master/snare/tanner_handler.py#L75) when detection is empty then raises a `keyError`. Since the code to handle is not written (which is because is unlikely to happen), that's why it's not present in the coverage. So I guess no harm in keeping this.

File: snare/tests/test_parse_tanner_response.py (hunk 6 above)
Comment (rnehra01) [reply snare/tests/test_parse_tanner_response.py]: Fix it.

**Comment by rnehra01** [comment]
> Coverage is at 99%. How should I test this [line](https://github.com/mushorg/snare/blob/c04984110c15d8fbe031a77d51af7f17a423ca5b/snare/tanner_handler.py#L58)?

I guess you can make `error()` a mock method and use `assert_called`.

File: snare/tests/test_parse_tanner_response.py (hunk 2 above)
Comment (rjt-gupta) [reply snare/tests/test_parse_tanner_response.py]: Okay,

**Comment by rjt-gupta** [comment]
Updated the changes.

File: snare/tests/test_submit_data.py [hunk 7]
```diff
@@ -6,10 +6,14 @@
 import json
 import yarl
 import aiohttp
+import logging
+from unittest import mock
 from snare.utils.asyncmock import AsyncMock
 from snare.tanner_handler import TannerHandler
 from snare.utils.page_path_generator import generate_unique_path
 
+logger = logging.getLogger(__name__)
```
Comment (rnehra01) [inline snare/tests/test_submit_data.py]: Can you place it in `setUp`?

File: snare/tests/test_submit_data.py [hunk 8]
```diff
@@ -65,6 +69,21 @@ def test_event_result(self):
         self.loop.run_until_complete(test())
         self.assertEqual(self.result, dict(detection={'type': 1}, sess_uuid="test_uuid"))
 
+    def test_submit_data_error(self):
+
+        async def mock_json_decode_error(data):
+            logger.error('Error submitting data: JSONDecodeError {}'.format(data))
+
+        self.handler = mock.Mock()
+        self.handler.submit_data = mock_json_decode_error
```
Comment (rnehra01) [inline snare/tests/test_submit_data.py]: I think you should use [this](https://github.com/mushorg/snare/blob/master/snare/tests/test_submit_data.py#L69) instead of mocking the entire `submit_data` method.

File: snare/tests/test_submit_data.py [hunk 9]
```diff
@@ -65,6 +69,21 @@ def test_event_result(self):
         self.loop.run_until_complete(test())
         self.assertEqual(self.result, dict(detection={'type': 1}, sess_uuid="test_uuid"))
 
+    def test_submit_data_error(self):
+
+        async def mock_json_decode_error(data):
+            logger.error('Error submitting data: JSONDecodeError {}'.format(data))
```
Comment (rnehra01) [inline snare/tests/test_submit_data.py]: Usually, logger creates a file for errors, is this also doing the same? If so please do the cleanup.

**Comment by rjt-gupta** [comment]
Done. 100% coverage reached.
